tst/unit/test_container_iterator.cpp

```diff
@@ -41,6 +41,39 @@ using parthenon::par_for;
 using parthenon::ParArrayND;
 using parthenon::Real;
 
+static void setVector(const ParArrayND<Real> &v, const Real &value) {
+  par_for(
+      "Initialize variables", DevExecSpace(), 0, v.GetDim(4) - 1, 0, v.GetDim(3) - 1, 0,
+      v.GetDim(2) - 1, 0, v.GetDim(1) - 1,
+      KOKKOS_LAMBDA(const int l, const int k, const int j, const int i) {
+        v(l, k, j, i) = value;
+      });
+}
+
+static Real sumVector(const ParArrayND<Real> &v) {
+  using policy4D = Kokkos::MDRangePolicy<Kokkos::Rank<4>>;
+  Real sum = 0.0;
+  Kokkos::parallel_reduce(
+      policy4D({0, 0, 0, 0}, {v.GetDim(4), v.GetDim(3), v.GetDim(2), v.GetDim(1)}),
+      KOKKOS_LAMBDA(const int l, const int k, const int j, const int i, Real &vsum) {
+        vsum += v(l, k, j, i);
```

Comment:
This is a no-no because you captured `ParArrayND` by reference, which means it's essentially a host-pointer. You need to either make a copy - something like `auto const v_ = v;`, or change the argument to the function to take `ParArrayND` by value rather than reference.

Reply:
Same as for `setVector`.

Reply:
Someday we'll have some clang tooling from the kokkos team that can catch this.

Reply:
Of course.  I was just trying to reduce duplicated code and didn't think through running on disjoint memory spaces.